advanced_functionality/causal-inference/ReadMe.md

```diff
@@ -0,0 +1,73 @@
+
+* __train__: The main program for training the model. When you build your own algorithm, you'll edit this to include your training code.
+* __serve__: The wrapper that starts the inference server. In most cases, you can use this file as-is.
+* __wsgi.py__: The start up shell for the individual server workers. This only needs to be changed if you changed where predictor.py is located or is named.
+* __predictor.py__: The algorithm-specific inference server. This is the file that you modify with your own algorithm's code.
+* __nginx.conf__: The configuration for the nginx master server that manages the multiple workers.
+
+### Setup for local testing
+
+The subdirectory local-test contains scripts and sample data for testing the built container image on the local machine. When building your own algorithm, you'll want to modify it appropriately.
+
+* __train-local.sh__: Instantiate the container configured for training.
+* __serve-local.sh__: Instantiate the container configured for serving.
+* __predict.sh__: Run predictions against a locally instantiated server.
+* __test-dir__: The directory that gets mounted into the container with test data mounted in all the places that match the container schema.
+* __payload.json__: Sample data for used by predict.sh for testing the server.
+
+#### The directory tree mounted into the container
+
+The tree under test-dir is mounted into the container and mimics the directory structure that SageMaker would create for the running container during training or hosting.
+
+* __input/data/training/heart_failure_clinical_records_dataset.csv__: The training data.
+* __model__: The directory where the algorithm writes the model file.
+* __output__: The directory where the algorithm can write its success or failure file.
+
+
+#### Local testing of the container
+To locally debug the container, without creating the endpoint, open one terminal and run:
+bash container/local_test/serve_local.sh causal-nex-container
+
+With open terminal, open second terminal and run:
+bash container/local_test/predict.sh *your_test_file.json*
+
+## Environment variables
+
+When you create an inference server, you can control some of Gunicorn's options via environment variables. These
+can be supplied as part of the CreateModel API call.
+
+    Parameter                Environment Variable              Default Value
+    ---------                --------------------              -------------
+    number of workers        MODEL_SERVER_WORKERS              the number of CPU cores
+    timeout                  MODEL_SERVER_TIMEOUT              60 seconds
+
+## Dataset
+Davide Chicco, Giuseppe Jurman: “Machine learning can predict survival of patients with heart failure from serum creatinine and ejection fraction alone”. BMC Medical Informatics and Decision Making 20, 16 (2020). 
+[Web Link]: https://bmcmedinformdecismak.biomedcentral.com/articles/10.1186/s12911-020-1023-5 
```

Comment:
Do you mean for these last few lines to be hyperlinks (lines 64-72)? Formatting is `[text](link)`